sn_client/src/wallet.rs

```diff
@@ -280,14 +298,38 @@ impl Client {
                 "sending spend request to the network: {:?}: {spend_request:#?}",
                 spend_request.unique_pubkey()
             );
-            tasks.push(self.network_store_spend(spend_request.clone(), verify_store));
+
+            let the_task = async move {
+                let cash_note_key = spend_request.unique_pubkey();
+                let result = self
+                    .network_store_spend(spend_request.clone(), verify_store)
+                    .await;
+
+                (cash_note_key, result)
+            };
+            tasks.push(the_task);
         }
 
-        for spend_attempt_result in join_all(tasks).await {
-            spend_attempt_result.map_err(|err| WalletError::CouldNotSendMoney(err.to_string()))?;
+        let mut spent_cash_notes = Vec::new();
+        for (cash_note_key, spend_attempt_result) in join_all(tasks).await {
+            // if this is a record mismatch on spend, we need to clean up and remove the spent CashNote from the wallet
```

Comment:
Hey Josh, I followed the flow there and even though we emit `ReturnedRecordDoesNotMatch` during `put_record_once(verify=true)`, that error is overwritten as shown below. Thus we might not be exposed to it.
```
// Verify the record is stored, requiring re-attempts
self.get_record_from_network(record_key, verify_store, GetQuorum::All, true)
    .await
    .map_err(|e| {
        error!(
            "Failing to verify the put record {:?} with error {e:?}",
            pretty_key
        );
        Error::FailedToVerifyRecordWasStored(pretty_key)
    })?;
```

Reply:
Ahhh.. I was not on the latest main 🤦 It works as intended there! You can safely ignore this comment!